pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -55,10 +48,17 @@ const (
 	// leaking a blob that is no longer tagged in given repository.
 	BlobRepositoryCacheTTLEnvVar = "REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_BLOBREPOSITORYCACHETTL"
 
+	// ManifestStorageEnvVar defines the storage type for manifests.
+	ManifestStorageEnvVar = "REGISTRY_MIDDLEWARE_MANIFEST_STORAGE"
+
 	// Default values
 
 	defaultDigestToRepositoryCacheSize = 2048
 	defaultBlobRepositoryCacheTTL      = time.Minute * 10
+
+	// Possible storages for manifests.
+	manifestStorageEtcd     = "etcd"
```

Comment:
I'm not sure this should be a config variable.  Why would admins want this choice?  The alternative is to always write manifests to storage, and on reads, look first in storage and only after look in etcd.